AsyncDisplayKit/ASCollectionView.h

```diff
@@ -319,15 +319,26 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * @param indexPath The index path of the requested node.
  *
- * @returns a node for display at this indexpath.  Must be thread-safe (can be called on the main thread or a background
- * queue) and should not implement reuse (it will be called once per row).  Unlike UICollectionView's version, this method
+ * @returns a node for display at this indexpath. This will be called on the main thread and should not implement reuse (it will be called once per row).  Unlike UICollectionView's version, this method
  * is not called when the row is about to display.
  */
 - (ASCellNode *)collectionView:(ASCollectionView *)collectionView nodeForItemAtIndexPath:(NSIndexPath *)indexPath;
 
+
 @optional
 
 /**
+ *
+ * @param collectionView The sender.
+ *
+ * @param indexPath The index path of the requested node.
+ *
+ * @returns a block that creates the node for display at this indexpath.  Must be thread-safe (can be called on the main thread or a background
+ *          queue) and should not implement reuse (it will be called once per row).
+ */
+- (ASDataControllerCellNodeBlock)collectionView:(ASCollectionView *)collectionView nodeBlockAtIndexPath:(NSIndexPath *)indexPath;
```

Comment:
How can we best handle the fact that nodeForItemAtIndexPath: should not be required if nodeBlockAtIndexPath is implemented?

Probably should just go to runtime assertions.

How intentional was it to leave out the word "item" in the name?  I'm supportive of this to unify the collection / table APIs, but bummed that it doesn't match the non-block method.